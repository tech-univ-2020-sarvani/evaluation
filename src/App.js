/* eslint-disable react/require-default-props */
import React from 'react';
import './App.css';
import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';
import PropTypes from 'prop-types';
import HomePage from './containers/HomePage';
import Genres from './containers/Genres';
import Songs from './containers/Songs';

function App(props) {
  const { testId } = props;
  return (
    <div data-testId={testId}>
      <Router>
        <Switch>
          <Route path="/">
            <HomePage />
          </Route>
          <Route path="/genres">
            <Genres />
          </Route>
          <Route path="/genres/:type">
            <Songs />
          </Route>
        </Switch>
      </Router>
    </div>
  );
}
App.propTypes = {
  testId: PropTypes.string,
};
export default App;
