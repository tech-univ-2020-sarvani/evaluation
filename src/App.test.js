import React from 'react';
import { render } from '@testing-library/react';
import App from './App';

describe('the App', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<App testId="test-app" />);

    expect(asFragment()).toMatchSnapshot();
  });
});
