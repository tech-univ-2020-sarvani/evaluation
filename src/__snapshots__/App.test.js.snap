// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the App should render correctly 1`] = `
<DocumentFragment>
  <div
    class="homePage"
  >
    <div
      class="NavBar"
    >
      <p>
        <b>
          Record
        </b>
        Shelf
      </p>
    </div>
    <div
      class="container"
    >
      <p>
        :((
      </p>
      <br />
      <h3>
        seems a bit empty in here..
      </h3>
      <div
        class="buttonContainer"
      >
        <button
          class="button"
          type="button"
        >
          sync
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`the Nav component should render correctly 1`] = `
<DocumentFragment>
  <div
    class="homePage"
  >
    <div
      class="NavBar"
    >
      <p>
        <b>
          Record
        </b>
        Shelf
      </p>
    </div>
    <div
      class="container"
    >
      <p>
        :((
      </p>
      <br />
      <h3>
        seems a bit empty in here..
      </h3>
      <div
        class="buttonContainer"
      >
        <button
          class="button"
          type="button"
        >
          sync
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
